### PR TITLE
Tidy up typing in OC\Files\View

### PR DESCRIPTION
**Pull request by come-nc**
## Summary

Add return types and fix psalm complaints in View and related classes.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by come-nc** [comment]
Needs https://github.com/icewind1991/Streams/pull/17

**Comment by come-nc** [comment]
@icewind1991 https://github.com/nextcloud/server/blob/master/apps/dav/lib/Connector/Sabre/Directory.php#L139-L141 this is blowing up because it puts null in non-nullable parameters.
Any idea how to refactor this properly?

**Comment by come-nc** [comment]
> @icewind1991 https://github.com/nextcloud/server/blob/master/apps/dav/lib/Connector/Sabre/Directory.php#L139-L141 this is blowing up because it puts null in non-nullable parameters. Any idea how to refactor this properly?

For now I went back to weak typing for internalPath in FileInfo.

**Review by ChristophWurst — COMMENTED** [review]
:+1: looks ok but I'm not familiar enough with this area of the code to given an approval

File: lib/private/Files/View.php [hunk 1]
```diff
@@ -192,74 +181,56 @@ public function getRelativePath($path) {
...
 		} else {
-			return null;
-		}
-	}
-
-	/**
-	 * @param string $path
-	 * @return string
-	 */
-	public function getLocalFolder($path) {
```
Comment (icewind1991) [inline lib/private/Files/View.php:255]: is deleting this intentional?

**Review by icewind1991 — APPROVED** [review]
Good work otherwise :+1:

**Comment by icewind1991** [comment]
> @icewind1991 https://github.com/nextcloud/server/blob/master/apps/dav/lib/Connector/Sabre/Directory.php#L139-L141 this is blowing up because it puts null in non-nullable parameters. Any idea how to refactor this properly?

probably makes most sense to change those `null`s to empty strings. But it probably makes more sense to do a that in a separate PR

File: lib/private/Files/View.php (hunk 1 above)
Comment (come-nc) [reply lib/private/Files/View.php:255]: Yes, the only place it was used was a test.

File: lib/private/Files/View.php [hunk 2]
```diff
@@ -1163,9 +1124,9 @@ private function basicOperation($operation, $path, $hooks = [], $extraParam = nu
 			}
 
 			$run = $this->runHooks($hooks, $path);
-			/** @var \OC\Files\Storage\Storage $storage */
 			[$storage, $internalPath] = Filesystem::resolvePath($absolutePath . $postFix);
 			if ($run and $storage) {
+				/** @var \OC\Files\Storage\Storage $storage */
```
Comment (blizzz) [inline lib/private/Files/View.php:1129]: Namespace not necessary, use statement already present

File: lib/public/Files/FileInfo.php [hunk 3]
```diff
@@ -250,7 +250,7 @@ public function getMountPoint();
 	/**
 	 * Get the owner of the file
 	 *
-	 * @return \OCP\IUser
+	 * @return ?\OCP\IUser
```
Comment (blizzz) [inline lib/public/Files/FileInfo.php:253]: should be added to critical changes (even if not a de-facto change to raise awareness)

**Review by blizzz — APPROVED** [review]

**Pull request merged**